syntax = "proto3";

// service
service ImageProcedure {
    rpc getPred(B64Image) returns (Prediction) {}
}

// input image, width, height
message B64Image {
    string b64image = 1;
    int32 width = 2;
    int32 height = 3;
}
// The request message containing the image.
message MsgRequest {
    bytes img = 1;
    int32 width = 2;
    int32 height = 3;
}

// The response message containing the reply
message MsgReply {
  int32 reply = 1;
}

// output prediction

message Prediction {
    string id = 1;
    bytes returnedImg = 2;
    string bb_x_coor = 3;
    string bb_y_coor = 4;
}

